### PR TITLE
bounds checking for one case in dataview

### PR DESCRIPTION
**Pull request by gordonwoodhull**
probably there are other places where this is needed, but this one
had a FIXME comment, and it helps diagnose problems we are seeing with
messages apparently being broken into frames/fragments

### CODE REVIEW

**Comment by gordonwoodhull** [comment]
@s-u, this helps diagnose the bounds problems we are seeing in https://github.com/att/rcloud/issues/1133 (clever breakpoints also help)

File: src/endian_aware_dataview.js [hunk 1]
```diff
@@ -102,12 +102,14 @@
                     this.buffer, this.offset + offset, this.buffer.byteLength);
             },
             view: function(new_offset, new_length) {
-                // FIXME Needs bounds checking
+                var ofs = this.offset + new_offset;
+                if(ofs + new_length > this.buffer.byteLength)
+                    throw new Error("Rserve.my_ArrayBufferView.view: bounds error: sgize: " +
```
Comment (cscheid) [inline src/endian_aware_dataview.js]: Really super-minor, but `sgize` is a typo, I assume?

Comment (gordonwoodhull) [inline src/endian_aware_dataview.js]: thanks, focus in the wrong emacs window.  fixed.

**Comment by cscheid** [comment]
Looks good. Thank you.

**Comment by gordonwoodhull** [comment]
Thanks!  We also use #22 in current rcloud, in order to communicate with compute and control processes.

**Comment by cscheid** [comment]
Ok.

I’ll fix the tabs over the weekend and merge.=

**Pull request merged**